src/impl_/wrap.rs

```diff
@@ -40,11 +41,44 @@ where
     T: IntoPy<PyObject>,
 {
     type Error = E;
-    fn wrap(self, py: Python<'_>) -> Result<Py<PyAny>, Self::Error> {
-        self.map(|o| o.into_py(py))
+    #[inline]
+    fn wrap(self) -> Result<T, Self::Error> {
+        self
     }
 }
 
+/// This is a follow-up function to `OkWrap::wrap` that converts the result into
+/// a `*mut ffi::PyObject` pointer.
+pub fn map_result_into_ptr<T: IntoPy<PyObject>>(
+    py: Python<'_>,
+    result: PyResult<T>,
+) -> PyResult<*mut ffi::PyObject> {
+    result.map(|obj| obj.into_py(py).into_ptr())
+}
+
+/// This is a follow-up function to `OkWrap::wrap` that converts the result into
+/// a safe wrapper.
+pub fn map_result_into_py<T: IntoPy<PyObject>>(
+    py: Python<'_>,
+    result: PyResult<T>,
+) -> PyResult<PyObject> {
+    result.map(|err| err.into_py(py))
+}
+
+/// Used to wrap the result of async `#[pyfunction]` and `#[pymethods]`.
+#[cfg(feature = "macros")]
+pub fn wrap_future<F, R, T>(future: F) -> crate::coroutine::Coroutine
```

Comment:
I would say it should not be moved, but it don't really matter because it will be reworked in the next PR